atomic_reactor/util.py

```diff
@@ -738,21 +746,44 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
 
         # Only compare prefix as response may use +prettyjws suffix
         # which is the case for signed manifest
-        response_h_prefix = response.headers['Content-Type'].rsplit('+', 1)[0]
+        try:
+            received_media_type = response.headers.get('Content-Type')
+        except KeyError:
+            # Guess content_type from contents
+            try:
+                encoding = guess_json_utf(response.content)
+                manifest = json.loads(response.content.decode(encoding))
+                received_media_type = manifest['mediaType']
+            except (ValueError,  # not valid JSON
+                    KeyError):   # no mediaType key
+                logger.warning("Unable to fetch media type: neither Content-Type header "
+                               "nor mediaType in output was found")
+
+        if not received_media_type:
+            continue
+
+        # set it to truthy value so that koji_import would know pulp supports these digests
+        digests[version] = True
```

Comment:
Right, but this line exists only to handle the case where the digest was guessed, right?